test/unit/server/bin/plugin/plugin.js

```diff
@@ -0,0 +1,130 @@
+var root = require('requirefrom')('');
+var plugin = root('src/server/bin/plugin/plugin');
+var expect = require('expect.js');
+var sinon = require('sinon');
+var installer = root('src/server/bin/plugin/pluginInstaller');
+var remover = root('src/server/bin/plugin/pluginRemover');
+var settingParser = root('src/server/bin/plugin/settingParser');
+
+describe('kibana cli', function () {
+
+  describe('plugin installer', function () {
+
+    describe('settings.action', function () {
+
+      var program = {
+        command: function () { return program; },
+        description: function () { return program; },
+        option: function () { return program; },
+        action: function (processCommand) {
+          processCommand();
+        }
+      };
+
+      beforeEach(function () {
+        sinon.stub(remover, 'remove');
+        sinon.stub(installer, 'install');
+      });
+
+      afterEach(function () {
+        remover.remove.restore();
+        installer.install.restore();
+        settingParser.parse.restore();
+      });
+
+      it('should call remove if settings.action is "remove"', function () {
+        sinon.stub(settingParser, 'parse', function () {
+          return {
+            action: 'remove'
+          };
+        });
+
+        plugin(program);
```

Comment:
In a scenario like we usually do:

  1. write a helper function (commonly called `setup()` or `init()`) to setup your test with args, rather than using  copy, paste, and edit to modify the setup behavior.
  2. or my *personal* favorite, write a function which defines the whole test for us. This still allows us to use args for customization, but it also gives us a chance to define an entire suite (`describe()`) just for this test, with all of the custom setup and teardown logic in legit `before()` and `after()` handlers.

    ```js
    function testUsageOfAction(action) {
      describe('when action is ' + action, function () {
        before(function () {
          sinon
          .stub(settingParser, 'parse')
          .returns({ action: action });

          plugin(program);
        });
    
        it('calls the right function', function () {
          expect(remover.remove.called).to.be(action === 'remove');
          expect(installer.install.called).to.be(action === 'install');
        });
    
        after(function () {
          settingParser.parse.restore();
        });
      });
    }
    
    testUsageOfAction('remove');
    testUsageOfAction('install');
    ```